runtime/vm/resolvefield.cpp

```diff
@@ -36,6 +36,9 @@
 /* Extra hidden fields are lockword and finalizeLink. */
 #define NUMBER_OF_EXTRA_HIDDEN_FIELDS 2
 
+/* Max hot field list length allowed for a class if scavenger dynamicBreadthScanOrdering is enabled */
+#define MAX_HOT_FIELD_LIST_LENGTH 10
```

Comment:
Might be worth making this a commandline param so it can be toggled if you feel that varying the number of hotfields may be preferable in certain apps